newbank/server/Commands/OfferCommand.java

```diff
@@ -0,0 +1,197 @@
+package newbank.server.Commands;
+
+import newbank.server.Account;
+import newbank.server.Currency;
+import newbank.server.Customer;
+import newbank.server.MicroLoanMarketPlace;
+import newbank.server.Offer;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class OfferCommand extends NewBankCommand {
+
+  private final double MAXOFFER = 2500;
+  private final int MAXLENGTH = 24;
+  private final double MAXRATE =
+      MicroLoanMarketPlace.getInstance().getMaxInterestRate().doubleValue();
```

Comment:
It would probably be better for all constants to be contained in one class rather than spread between here and MicroLoanMarketPlace. I can't think of a good reason to access those values outside of Offer so it would probably be best to move them all to Offer and delete the getMaxInterestRate method.

Reply:
Hi, yes, I wasn't sure whether they should go int the market class or offer command class, so left them half and half. I have now put them all in the offer command class